### PR TITLE
feat: update base docker image from node 12 to node 18

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

**Review by asottile-sentry — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -24,7 +24,10 @@ RUN export YARN_CACHE_FOLDER="$(mktemp -d)" \
 # Copy the artifacts from `yarn build`
 COPY --from=builder /app/dist /action-release/dist/
 RUN chmod +x /action-release/dist/index.js
-ENV SENTRY_BINARY_PATH=/action-release/node_modules/.bin/sentry-cli
+
+# move the sentry-cli binary to where the entrypoint expects it
+RUN mv /action-release/node_modules/@sentry/cli/sentry-cli /action-release/sentry-cli
+RUN chmod +x /action-release/sentry-cli
```
Comment (asottile-sentry) [inline Dockerfile:30]: this isn't right -- something should be adjusted to find it in the original place

File: Dockerfile [hunk 2]
```diff
@@ -25,6 +25,10 @@ RUN export YARN_CACHE_FOLDER="$(mktemp -d)" \
 COPY --from=builder /app/dist /action-release/dist/
 RUN chmod +x /action-release/dist/index.js
 
+# move the sentry-cli binary to where the entrypoint expects it
+RUN mv /action-release/node_modules/@sentry/cli/sentry-cli /action-release/sentry-cli
```
Comment (scefali) [inline Dockerfile:29]: This chnage is temporary until we update the version of the CLI and can pass in an env variable to point to the CLI where it exists

Comment (armenzg) [inline Dockerfile:29]: How do you know the exact path?

File: src/main.ts [hunk 3]
```diff
@@ -82,6 +82,6 @@ import * as process from 'process';
     core.debug(`Done`);
     core.setOutput('version', version);
   } catch (error) {
-    core.setFailed(error.message);
+    core.setFailed((error as Error).message);
```
Comment (armenzg) [inline src/main.ts:85]: What's this for?

File: package.json
Comment (armenzg) [inline package.json:1]: Are you just updating all the dev dependencies?

File: Dockerfile (hunk 2 above)
Comment (scefali) [reply Dockerfile:29]: @armenzg that's just where it gets installed when we install the CLI package, I just found it

File: src/main.ts (hunk 3 above)
Comment (scefali) [reply src/main.ts:85]: @armenzg TS was yelling at me for this error, need to cast as an error or else compilation fails

File: package.json
Comment (scefali) [reply package.json:1]: @armenzg Yes, I needed to do that to get Node 18 to work. I tried to do it separately but alas not possible.

**Pull request merged**